marimo/_server/session/serialize.py

```diff
@@ -201,7 +222,9 @@ def serialize_notebook(
     """Convert a SessionView to a Notebook schema."""
     cells: list[NotebookCell] = []
 
-    for cell_id in view.cell_operations.keys():
+    # Use document order from cell_manager instead of execution order from session_view
+    # to ensure cells appear in the correct sequence in HTML export
+    for cell_id in cell_manager.cell_ids():
```

Comment:
Could this cause some reconciliation issue in the front end?

Reply:
This looks correct to me, since the notebook snapshot is only used used in the HTML export.